upup/pkg/fi/cloudup/alitasks/disk.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alitasks
+
+import (
+	"fmt"
+
+	"github.com/denverdino/aliyungo/common"
+	"github.com/denverdino/aliyungo/ecs"
+	"github.com/golang/glog"
+
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/aliup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+)
+
+// Disk represents a ALI Cloud Disk
+//go:generate fitask -type=Disk
+const (
+	DiskResource = "disk"
+	DiskType     = ecs.DiskTypeAllData
```

Comment:
just curious, what does ecs stand for?

Reply:
ecs stands for Elastic Computing Engine, the EC2 for AliCloud.
Most of the product APIs are implemented in ecs package.